openassessment/workflow/models.py

```diff
@@ -353,11 +422,27 @@ def set_score(self, score):
                 'points_possible'.
 
         """
-        sub_api.set_score(
-            self.submission_uuid,
-            score["points_earned"],
-            score["points_possible"]
-        )
+        if not self.staff_score_exists():
```

Comment:
@andy-armstrong - I now understand what you mean here a little better I think. It's not strictly necessary for this check to be here, the workflow `get_score()` function will still be working correctly due to the staff API coming first in priority order.

However, by keeping this check and using `reset_score` in `set_staff_score` above, we ensure that the score reported by the submissions API remains correct in the event that a different consumer asks for the current score on a given submission.